cms/envs/common.py

```diff
@@ -969,6 +970,7 @@
     XModuleMixin,
     EditInfoMixin,
     AuthoringMixin,
+    StudioEditableBlock,
```

Comment:
Is this allowed? this was the only way i could make sure all blocks would have the required callbacks

Reply:
@DanielVZ96 Great question.

I would rather not add StudioEditableBlock to this list, as that would make render_children and get_preview_view_name available on every XBlock. I'd like to avoid adding methods to the base XBlock API except in the cases where we really need to (like editor_saved, post_editor_saved, studio_post_duplicate).

That said, you are right that we need some way of defining editor_saved, post_editor_saved, and studio_post_duplicate on every block. I was mistaken that we could just add them to StudioEditableBlock. Could I suggest adding them to AuthoringMixin instead, which is already mixed into all CMS blocks?

Reply:
Done! thanks for the suggestion